File: app/src/main/java/com/example/testvolley/services/rickAndMorty/dtos/Page.java
package com.example.testvolley.services.rickAndMorty.dtos;

import android.content.Context;

import org.jetbrains.annotations.NotNull;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.Objects;

public class Page {

    private Informacion info;
    private Resultado results;
    private JSONObject informacion;

    public Page(@NotNull JSONObject jsonObject, Context context) throws JSONException {
        informacion = jsonObject.getJSONObject("info");
        //Toast.makeText(context, informacion.toString(), Toast.LENGTH_SHORT).show();
        this.info = new Informacion(Objects.requireNonNull(jsonObject.getJSONObject("info")));
        //this.results = new Resultado(Objects.requireNonNull(jsonObject.optJSONObject("results")));
    }


    public Informacion getInfo() {
        return info;
    }

    public void setInfo(Informacion info) {
        this.info = info;
    }

    public Resultado getResults() {
        return results;
    }

    public void setResults(Resultado results) {
        this.results = results;
    }



}

File: app/src/main/java/com/example/testvolley/services/rickAndMorty/dtos/Page.kt
package com.example.testvolley.services.rickAndMorty.dtos

import Informacion
import android.widget.Toast

class Page constructor(data : org.json.JSONObject) {

    lateinit var info : Informacion
    lateinit var results : org.json.JSONArray
    val personajes : MutableList<Personaje> = mutableListOf()

    init {
        info = Informacion(data.getJSONObject("info"))
        results = data.getJSONArray("results")
        init()
    }

    private fun init(){
        val nResult = results.length()
        for (i in 0 until results.length()){
            this.personajes.add(Personaje(results.getJSONObject(i)))
        }
    }




}